button.dust

```diff
@@ -0,0 +1,13 @@
+    {?element}
+        {element|s}
+    {:else}
+        {?href}
+            <a href={href} {?id}id="{id}"{/id} class="c-button {class} {?disabled}c-button__is-disabled{/disabled}" >
+                <span>{?text}{text}{/text}</span>
+            </a>
+        {:else}
+            <button type="button" {?id}id="{id}"{/id} class="c-button {class}" {?disabled}disabled="disabled"{/disabled}>
+                <span>{?text}{text}{/text}</span>
+            </button>
+        {/href}
+    {/element}
```

Comment:
I’d recommend a number of changes here:

- The `{element|s}` usage looks like it comes from the Select component. For that component, I think it’s sensible to support the use case of providing the entire <select> markup including options in one chunk (`element`). Here, that doesn’t make sense, since we’re creating our own button or anchor tag. All we need is the content between `{:else}` and `{/element}`.

- We should use the main component body instead of the `text` parameter for the button content, since that will be a more natural equivalent to what you’d write in HTML (we want to be able to write `{@c-button}Hello World{/c-button}`).

- I’d like to move towards standardizing the order of common html attributes, with `class` first, followed by `id`.

- The disabled state class should be `c--is-disabled` since it’s a modifier rather than a sub-component. In addition, the class should be applied to both the button and anchor elements so that they can be styled using the same selector.

That would leave us with the following template:

```html
{?href}
    <a class="c-button {class} {?disabled}c--is-disabled{/disabled}"
        {?id}id="{id}"{/id}
        href="{href}"
    >
        <span>{body}</span>
    </a>
{:else}
    <button class="c-button {class} {?disabled}c--is-disabled{/disabled}"
        {?id}id="{id}"{/id}
        type="{?type}{type}{:else}button{/type}"
        {?disabled}disabled{/disabled}
    >
        <span>{body}</span>
    </button>
{/href}
```